### PR TITLE
Implement parachain and validator registration process in Subxt

### LINKED ISSUE #106: Implement parachain and validator registration process in Subxt
This task involves registering a parachain using Subxt and registering a validator with Subxt as well. It's crucial to perform version validation checks on the calls, following the node version of Zombienet.

### PR DESCRIPTION
**Pull request by wirednkod**
Fixes #106

### CODE REVIEW

File: crates/orchestrator/src/network/parachain.rs [hunk 1]
```diff
@@ -35,4 +42,65 @@ impl Parachain {
...
+        // get the nonce for the sudo account
+        let account_nonce_call = subxt::dynamic::runtime_api_call(
+            "AccountNonceApi",
+            "account_nonce",
+            vec![account_id_value.clone()],
+        );
+
+        let nonce = api
+            .runtime_api()
+            .at_latest()
```
Comment (pepoviola) [inline crates/orchestrator/src/network/parachain.rs:83]: Do we need `at_latest`, let me double check the docs form subxt :)

File: crates/orchestrator/src/network/parachain.rs [hunk 2]
```diff
@@ -35,4 +42,65 @@ impl Parachain {
...
+            .await?
+            .call(account_nonce_call)
+            .await?;
+
+        println!("Account nonce: {:#?}", nonce.to_value());
+
+        let schedule_para = subxt::dynamic::tx(
```
Comment (pepoviola) [inline crates/orchestrator/src/network/parachain.rs:90]: Nice!

File: crates/orchestrator/src/shared/types.rs [hunk 3]
```diff
@@ -55,3 +58,39 @@ pub struct ChainDefaultContext<'a> {
...
+    }
+}
+
+impl Deref for ParachainGenesisArgs {
+    type Target = String;
+
+    fn deref(&self) -> &Self::Target {
+        &self.genesis_head
+    }
+}
```
Comment (pepoviola) [inline crates/orchestrator/src/shared/types.rs:96]: Do we need this? why are you deref to the genesis_head?

Comment (wirednkod) [reply crates/orchestrator/src/shared/types.rs:96]: The `subxt::dynamic::tx` receives  the `call_data` as a `vec![]` of `Values::from_bytes`...
The call is:
```rs
let schedule_para = subxt::dynamic::tx(
            "ParasSudoWrapperCall",
            "sudo_schedule_para_initialize",
            vec![
                Value::from_bytes(account_id),
                Value::from_bytes(parachain_genesis_value), // <--- custom type `ParachainGenesisArgs`
            ],
        );
```

In order to satisfy the the trait bound `ParachainGenesisArgs: AsRef<[u8]>`  needs to be implemeted but I am not sure how to Deref for the whole struct. 
For a temp soluton I left the string of `genesis_head` - but a correct struct should return

**Comment by pepoviola** [comment]
I will merge this one as is now, and the refactor a little bit and add the validator register/desregister.
Thx!

**Review by pepoviola — APPROVED** [review]

**Pull request merged**